rs/replicated_state/src/canister_state/system_state/call_context_manager.rs

```diff
@@ -784,12 +796,34 @@ impl CallContextManager {
     pub fn unregister_callback(&mut self, callback_id: CallbackId) -> Option<Arc<Callback>> {
         self.callbacks.remove(&callback_id).map(|callback| {
             self.stats.on_unregister_callback(&callback);
+            if callback.deadline != NO_DEADLINE {
+                self.unexpired_callbacks
+                    .remove(&(callback.deadline, callback_id));
+            }
             debug_assert!(self.stats_ok());
 
             callback
         })
     }
 
+    /// Returns the IDs of all best-effort callbacks whose deadlines have expired
+    /// since the previous call to this method, given the current time.
+    ///
+    /// Note: A given callback will be returned at most once by this function.
+    #[allow(dead_code)]
+    pub(super) fn expired_callbacks(
```

Comment:
Nit: the name `expired_callbacks` suggests that self would not be modified. I think something like `expire_callbacks` (or something similar) would be better to make sure it is clear that self is mutated.

Reply:
Good point, done.